opentracing-cdi-example/src/main/java/io/opentracing/contrib/cdi/example/SpanContextProducer.java

```diff
@@ -0,0 +1,18 @@
+package io.opentracing.contrib.cdi.example;
+
+import io.opentracing.SpanContext;
+import io.opentracing.contrib.web.servlet.filter.TracingFilter;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+
+public class SpanContextProducer {
+    @Inject
+    HttpServletRequest request;
+
+    @Produces
+    public SpanContext produceSpanContext() {
+        return (SpanContext) request.getAttribute(TracingFilter.SERVER_SPAN_CONTEXT);
```

Comment:
Note that this is on the example code, this is not part of the CDI integration. I'm not sure we can (or should) make the assumption that the span context is what we have on the request. Should we?